### PR TITLE
Reorganize assembly definitions and directory structure of SpectatorView.Unity

### PR DESCRIPTION
**Pull request by matthejo**
This change reorganizes SpectatorView.Unity into new assembly definitions, and will standardize the namespaces across the project.

### CODE REVIEW

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Sharing/ArUcoMarkerVisualSpatialLocalizer.cs [hunk 1]
```diff
@@ -1,15 +1,15 @@
-﻿using Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.MarkerDetection;
-using Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.Utilities;
-using System;
-
-namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
-{
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Sharing/ArUcoMarkerVisualSpatialLocalizer.cs]: do we want to update namespaces in this review as well?

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/MarkerDetectorCoordinateService.cs [hunk 2]
```diff
@@ -1,190 +1,190 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    /// <summary>
+    /// A marker detection based implementation of <see cref="ISpatialCoordinateService"/>.
+    /// </summary>
+    public class MarkerDetectorCoordinateService : SpatialCoordinateServiceBase<int>
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/MarkerDetectorCoordinateService.cs]: >MarkerDetectorCoordinateService [](start = 17, length = 31)

nit: additional subfolders may help here for clarity. Some potential examples:

SpatialAlignment/MarkerVisual
SpatialAlignment/PhysicalMarker
SpatialAlignment/Persistence

**Comment by chrisfromwork** [comment]
> 

nit: this class can be deleted, calibration uses the SocketerNetworkingService.

---
Refers to: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/UDPBroadcastNetworkingService.cs:23 in b3ab332. [](commit_id = b3ab3325016045211f412b084e12993498502161, deletion_comment = False)

File: src/SpectatorView.Unity/Assets/SpatialAlignment/Scripts/Utilities/SpatialCoordinateLocalizer.cs [hunk 3]
```diff
@@ -1,144 +1,144 @@
...
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Experimental.SpatialAlignment.Common
+{
+    /// <summary>
+    /// Very simple consumer of <see cref="ISpatialCoordinate"/> to demonstrate usage.
+    /// </summary>
+    public class SpatialCoordinateLocalizer : MonoBehaviour
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpatialAlignment/Scripts/Utilities/SpatialCoordinateLocalizer.cs]: >SpatialCoordinateLocalizer [](start = 17, length = 26)

nit: spectator view doesn't currently use this file. If we want ot keep it, it should probably go the in the sharing.spatialalignment directories.

File: src/SpectatorView.Unity/Assets/SpatialAlignment/Scripts/Utilities/Extensions.cs [hunk 4]
```diff
@@ -1,237 +1,237 @@
...
+{
+    /// <summary>
+    /// Helper Unity side extensions to enable the core abstraction to be Unity agnostic.
+    /// </summary>
+    public static class Extensions
+    {
+        /// <summary>
+        /// Converts a <see cref="Vector3"/> to a <see cref="UVector3"/>.
+        /// </summary>
+        public static UVector3 AsUnityVector(this Vector3 input)
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpatialAlignment/Scripts/Utilities/Extensions.cs]: >AsUnityVector [](start = 31, length = 13)

nit: i think andrei had hoped for these extension utilities to remain in the spatial alignment repo.

File: src/SpectatorView.Unity/Assets/SpatialAlignment/Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing.asmdef [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing",
+    "name": "Microsoft.MixedReality.SpatialAlignment",
     "references": [
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpatialAlignment/Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing.asmdef]: nit: we should rename this file to match the new dll name.

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/ISpatialCoordinateServiceOld.cs [hunk 6]
```diff
@@ -1,38 +1,38 @@
...
+using UnityEngine;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    /// <summary>
+    /// Delegate called when a new spatial coordinate state payload has been created
+    /// </summary>
+    /// <param name="payload"></param>
+    public delegate void SpatialCoordinateStateUpdateHandler(byte[] payload);
+
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpectatorView/Scripts/ISpatialCoordinateServiceOld.cs]: nit: i think we can delete this file now

**Comment by chrisfromwork** [comment]
>     {

nit: i think we can delete this file now.

---
Refers to: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpectatorViewOld.cs:10 in 1175d6d. [](commit_id = 1175d6d2b32264aa4732c8b4da99d14b38a55840, deletion_comment = False)

**Comment by chrisfromwork** [comment]
>     public class SceneRoot : MonoBehaviour

nit: i think we can delete this file now.

---
Refers to: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SceneRoot.cs:12 in 1175d6d. [](commit_id = 1175d6d2b32264aa4732c8b4da99d14b38a55840, deletion_comment = False)

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/ISpatialLocalizer.cs [hunk 7]
```diff
@@ -1,26 +1,27 @@
...
+using UnityEngine;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    public interface IPeerConnection
+    {
+        void SendData(Action<BinaryWriter> writeCallback);
+    }
+
+    public interface ISpatialLocalizer
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/ISpatialLocalizer.cs]: >ISpatialLocalizer [](start = 21, length = 17)

Do we know whether we want ISpatialLocalizer and ISpatialLocalizationSettings to be maintained in the SpatialAlignment repo/logic?

Comment (matthejo) [reply src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/ISpatialLocalizer.cs]: The current plan is to move this code as part of a future task.

**Pull request merged**